src/pages/ProviderEServiceCreatePage/components/EServiceCreateStepPurpose/EServiceCreateFromTemplateStepPurpose.tsx

```diff
@@ -0,0 +1,103 @@
+import { SectionContainer, SectionContainerSkeleton } from '@/components/layout/containers'
+import { Button, Stack } from '@mui/material'
+import React, { useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { useEServiceCreateContext } from '../EServiceCreateContext'
+import { RHFAutocompleteSingle } from '@/components/shared/react-hook-form-inputs'
+import { FormProvider, useForm } from 'react-hook-form'
+import { StepActions } from '@/components/shared/StepActions'
+import ArrowBackIcon from '@mui/icons-material/ArrowBack'
+import ArrowForwardIcon from '@mui/icons-material/ArrowForward'
+import { EServiceRiskAnalysisInfoSummary } from '@/components/shared/RiskAnalysisInfoSummary'
+
+type RiskAnalysisValue = { label: string; value: string }
+
+export const EServiceCreateFromTemplateStepPurpose: React.FC = () => {
+  const { t } = useTranslation('eservice', { keyPrefix: 'create' })
+  const { t: tCommon } = useTranslation('common')
+
+  const { descriptor, forward, back } = useEServiceCreateContext()
+
+  const methods = useForm()
+
+  const riskAnalysisList = descriptor?.eservice.riskAnalysis ?? []
+
+  const autocompleteOptions = (riskAnalysisList ?? []).map((riskAnalysis) => ({
+    label: riskAnalysis.name,
+    value: riskAnalysis.id,
+  }))
+
+  const [selectedRiskAnalysis, setSelectedRiskAnalysis] = useState<RiskAnalysisValue>({
+    label: '',
+    value: '',
+  })
+  const [showRiskAnalysis, setShowRiskAnalisys] = useState(false)
+
+  const resetStates = () => {
+    setShowRiskAnalisys(false)
+    setSelectedRiskAnalysis({ label: '', value: '' })
+  }
+
+  const setStates = (value: RiskAnalysisValue) => {
+    setShowRiskAnalisys(false)
+    setSelectedRiskAnalysis(value)
+  }
+
+  return (
+    <FormProvider {...methods}>
+      <SectionContainer
+        title={t('stepPurpose.purposeTableSection.title')}
+        description={t('stepPurpose.purposeTableSection.descriptionEServiceFromTemplate')}
+      >
+        <Stack spacing={2} alignItems="flex-start">
+          <RHFAutocompleteSingle
+            sx={{ my: 0 }}
+            name="riskAnalysis"
+            label={t('stepPurpose.purposeTableSection.labelAutocompleteEServiceFromTemplate')}
+            options={autocompleteOptions}
+            onValueChange={(value) => (value === null ? resetStates() : setStates(value))}
+            rules={{ required: true }}
+          />
+          <Button
+            variant="contained"
+            disabled={selectedRiskAnalysis.value === ''}
+            onClick={() => setShowRiskAnalisys(true)}
```

Comment:
Since we are inside a form, the action of this button should be handled in a submit action